iceoryx_posh/source/popo/ports/application_port.cpp

```diff
@@ -23,20 +24,27 @@ ApplicationPort::ApplicationPort(ApplicationPortData* const applicationPortDataP
 {
 }
 
-bool ApplicationPort::dispatchCaProMessage(const capro::CaproMessage& message) noexcept
+cxx::optional<capro::CaproMessage> ApplicationPort::getCaProMessage() noexcept
 {
-    return getMembers()->m_caproMessageFiFo.push(message);
+    auto maybeMessage = getMembers()->m_caproMessageFiFo.pop();
```

Comment:
You could also write: 
```
return getMembers()->m_caproMessageFiFo.pop();
```

and get rid off the additional if else.

Reply:
good point, I somehow had in mind that the FiFo has no optional but a bool return.